#!/usr/bin/env python
PACKAGE = "stargazer_ros_tool"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("tight_filter_size", int_t, 0, "Size of the tighter box filter of DOB", 0, 0, 255)
gen.add("wide_filter_size", int_t, 0, "Size of the wider box filter of DOB", 0, 0, 255)
gen.add("threshold", int_t, 0, "Grayvalue threshold for landmark detection", 0, 0, 255)
gen.add("maxRadiusForPixelCluster", double_t, 0, "Maximum radius for clustering of pixels to points", 0, 0, 2000)
gen.add("maxRadiusForCluster", double_t, 0, "Maximum radius for clustering of points to landmarks", 0, 0, 2000)
gen.add("minPixelForCluster", int_t, 0, "Minimum count of pixels per point", 0, 0, 2000)
gen.add("maxPixelForCluster", int_t, 0, "Maximum count of pixels per point", 0, 0, 2000)
gen.add("minPointsPerLandmark", int_t, 0, "Minimum count of points per landmark", 0, 0, 2000)
gen.add("maxPointsPerLandmark", int_t, 0, "Maximum count of points per landmark", 0, 0, 2000)
gen.add("debug_mode", bool_t, 0, "Activate debug mode for landmark finder to show results", False)

exit(gen.generate(PACKAGE, "stargazer_ros_tool", "LandmarkFinder"))
